### ISSUE TITLE
Get different results after save and load the seq2seq model

### ISSUE THREAD

**Issue by sun-peach** [keras-team/keras#13290]
**System information**  
- Have I written custom code (as opposed to using example directory):  
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  Ubuntu 16.04
- TensorFlow backend (yes / no):  yes
- TensorFlow version:   1.13
- Keras version:  2.2.5
- Python version:  3.5
- CUDA/cuDNN version: 10.0 
- GPU model and memory:  2080 Ti

Hi, dear all,

I am building a seq2seq model. The core part is listed as following:

**First, I have a RNN as an encoder:**

`model_in = Input(shape=(500,40))`
`temp_value = Masking()(model_in)`
`temp_value = bidirectional(LSTM(128))(temp_value)`
`encoder_inputs = bidirectional(LSTM(128))(temp_value)`
`encoder = Model(inputs=model_in, outputs=encoder_in,name="Encoder")`

**Second I setup a step model for decoder, which can take one input and output one step:**
`# setup step inputs`
`states_pre = [Input(shape=(attend_dmodel,),name="H_pre"), Input(shape=(attend_dmodel,),name="C_pre")]`
`context_pre = Input(shape=(1,encoder_inputs.shape[-1].value),name="Context_pre")`
`step_input = Input(shape=(1,label_inputs.shape[-1].value),name='Y_pre')`

`query_add_time = Reshape((1,128))`
`attend_remove_time = Reshape((int(128,))`
`# warp single step model`
`# decoder LSTM input (concatenation of y_i-1 and c_i-1)`
`decoder_lstm_in = embedding_concatenate([step_input,context_pre])`
`# get state from lstm (no output needed, because it is equivilent to the state_h for single step)`
`_, state_h, state_c = decoder_lstm(decoder_lstm_in,initial_state=states_pre)`
`# use state_h as query for multi-head-attention`
`decode_query = query_add_time(state_h)        # add a dummy time dimension (length 1 time series)`
`# multi-head attend`
`c = attend_layer([decode_query, encoder_inputs, encoder_inputs, length_inputs])`
`c = non_mask(c)`
`context_embedding = c       # save c_i for input i+1`
`c = attend_remove_time(c)        # remove dummy time dimension (form a vector)`
`# generate spelling result`
`generate_tensor = generate_concatenate([state_h,c])`
`generate_tensor = Dense(128,activation="relu")(generate_tensor)`
`generate_tensor = generate_add_time(generate_tensor)        # add a dummy time dimension (length 1 time series)`
`step_output = Dense(31,activation="softmax")(generate_tensor)`
`step_input_list = [encoder_inputs,step_input,length_inputs,states_pre[0],states_pre[1],context_pre]`
`step_output_list = [step_output, state_h,state_c,context_embedding]`
`step_model = Model(inputs=step_input_list, outputs=step_output_list,name="Decoder_step")`

**Third I use a loop to call this step model, to take a sequence in and generate a sequence out:**
`all_outputs = []`
`label_inputs = Input(shape=(50,31))`
`y_in = splice2D(name="Splice2D_Init")(label_inputs)`
`context_embedding = context_pre`
`states = states_pre`
`for p_i in range(50):`
    `outputs = step_model([encoder_inputs,y_in,length_inputs,states[0],states[1],context_embedding])`
   ` y_out = outputs[0]`
   ` states = [outputs[1],outputs[2]]`
    `context_embedding = outputs[3]`
    `# Store the current prediction (we will concatenate all predictions later)`
    `all_outputs.append(y_out)`
    `# Reinject the outputs as inputs for the next loop iteration`
    `curr_label_input = splice2D(start=p_i+1)(label_inputs)`
    `y_in = y_out`
`seq_output = Lambda(lambda x: K.concatenate(x, axis=1),name=model_output_name)(all_outputs)`
`input = [encoder_inputs,label_inputs,length_inputs,states_pre[0],states_pre[1],context_pre]`
`output = seq_output`
`return_model = Model(inputs=input, outputs=output,name=model_name)`

Then I train and test this model, everything is fine (without saving, quiting and loading). But when I save, quit and reload it. I am not able to replicate the performance I had before (gap is huge). 
After quit and load, the performance is like random guess. Can anyone tell me what is wrong?

Thank you very much!
